src/response.rs

```diff
@@ -64,7 +122,58 @@ impl Response {
         }
     }
 
-    /// Get the `Headers`.
+    /// Get the `Headers` of this `Response`.
+    ///
+    /// ```rust
+    /// # extern crate reqwest;
+    /// # #[macro_use] extern crate error_chain;
+    /// # use std::fs;
+    /// # use std::io::{Read, Write};
+    /// # use reqwest::{Client, StatusCode};
+    /// # use reqwest::header::ContentType;
+    /// # use reqwest::mime::{Mime, TopLevel, SubLevel};
+    /// # error_chain! {
+    /// #     foreign_links {
+    /// #         Reqwest(reqwest::Error);
+    /// #         Io(::std::io::Error);
+    /// #     }
+    /// # }
+    /// # fn run() -> Result<()> {
+    /// let client = Client::new()?;
+    /// let mut resp = client.get("https://img.shields.io/crates/v/mime.png?label=mime")?.send()?;
+    /// match *resp.status() {
+    ///     StatusCode::Ok => (),
+    ///     status => bail!("Bad http response, status: {:?}", status),
+    /// };
+    ///
+    /// {
+    ///     let content_type = resp.headers().get::<ContentType>();
+    ///     if let Some(ct) = content_type {
+    ///         match **ct {
+    ///             Mime(TopLevel::Image, SubLevel::Png, _) => {
```

Comment:
I've released a v0.3, which unfortunately we can't really use yet, since it'd require breaking hyper, which can't happen. It's in the new (not yet released) 0.11 of hyper.

Basically you can either compare with the mime directly, or via parts. So, either `content_type.0 == mime::IMAGE_PNG`, if you wanted an exact match, or with a match if you wanted to ignore parameters or something:

```rust
match (content_type.type_(), content_type.subtype()) {
    (mime::IMAGE, mime::PNG) => (),
    _ => (),
}
```

No more importing all the different types. But as I said, we can't change this yet, so it's more just me complaining to myself...